src/enableDisableRules.ts

```diff
@@ -63,10 +72,46 @@ export class EnableDisableRulesWalker extends SkippableTokenAwareRuleWalker {
                     if (!(ruleToAdd in this.enableDisableRuleMap)) {
                         this.enableDisableRuleMap[ruleToAdd] = [];
                     }
-                    this.enableDisableRuleMap[ruleToAdd].push({
-                        isEnabled: isEnabled,
-                        position: startingPosition,
-                    });
+                    if (isCurrentLine) {
+                        // start at the line beginning
+                        this.enableDisableRuleMap[ruleToAdd].push({
+                            isEnabled: isEnabled,
+                            position: lineStartingPosition,
+                        });
+                        // end at the end of the comment
+                        this.enableDisableRuleMap[ruleToAdd].push({
+                            isEnabled: !isEnabled,
+                            position: scanner.getTextPos() + 1,
+                        });
+                    } else {
+                        this.enableDisableRuleMap[ruleToAdd].push({
+                            isEnabled: isEnabled,
+                            position: startingPosition,
+                        });
+
+                        if (isNextLine) {
+                            const nextLineEndPos = scanner.lookAhead(() => {
```

Comment:
I think you should be able to simplify this by passing in the `SourceFile` node and using `node.getPositionOfLineAndCharacter` or maybe `node.getLineStarts`